### PR TITLE
chore(developer): Refactor/developer/8086 common includes

### LINKED ISSUE #8086: chore(developer): check for common includes in CheckfilenameConsistency.cpp
In `developer/src/kmcmplib/src/CheckFilenameConsistency.cpp` (l. 116)

`//TODO: sort out how to find common includes in non-Windows platforms:`

we may need to address this. 

see    https://github.com/keymanapp/keyman/pull/6980#discussion_r1063292099  in feat(developer): cross-platform kmx compiler 🚑 https://github.com/keymanapp/keyman/pull/6980

### PR DESCRIPTION
**Pull request by SabineSIL**
Fixes #8086.

In developer/src/kmcmplib/src/CheckFilenameConsistency.cpp (l. 116)
//TODO: sort out how to find common includes in non-Windows platforms:  

Change lower part of the function CheckFilenameConsistency() to enable the use of CHINT_FilenameHasDifferingCase-Warnings on non-windows platforms as well

see Issue #8086 and PR #7330   ( PR #8335 )
@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmcmplib/src/CheckFilenameConsistency.cpp [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 #include "compfile.h"
 #include <comperr.h>
 #include "kmcmpdll.h"
 #include <io.h>
 #include <string>
 #include "CheckFilenameConsistency.h"
 #include "kmx_u16.h"
+#include <experimental/filesystem>
+namespace fs = std::experimental::filesystem;
+using std::experimental::filesystem::directory_iterator;
```
Comment (srl295) [inline developer/src/kmcmplib/src/CheckFilenameConsistency.cpp:13]: C++17, is that OK?

Comment (srl295) [reply developer/src/kmcmplib/src/CheckFilenameConsistency.cpp:13]: windows/src/engine/testhost/testhost.cpp just uses `#include <filesystem>`

Comment (srl295) [reply developer/src/kmcmplib/src/CheckFilenameConsistency.cpp:13]:
```suggestion
#include <filesystem>
namespace fs = std::filesystem;
using fs::directory_iterator;
```

File: developer/src/kmcmplib/src/CheckFilenameConsistency.cpp [hunk 2]
```diff
@@ -1,13 +1,16 @@
 
+#define _SILENCE_EXPERIMENTAL_FILESYSTEM_DEPRECATION_WARNING 1
```
Comment (srl295) [inline developer/src/kmcmplib/src/CheckFilenameConsistency.cpp:2]:
```suggestion
```

File: developer/src/kmcmplib/src/CheckFilenameConsistency.cpp (hunk 1 above)
Comment (SabineSIL) [reply developer/src/kmcmplib/src/CheckFilenameConsistency.cpp:13]: I wanted to do exactly your solution but this was not possible therefore I changed it to experimental/... and set the DEPRECATION_WARNING

**Pull request merged**